app/selectors/accountsController.ts

```diff
@@ -62,11 +70,18 @@ export const selectSelectedInternalAccount = createDeepEqualSelector(
 /**
  * A memoized selector that returns the selected internal account address in checksum format
  */
-export const selectSelectedInternalAccountChecksummedAddress = createSelector(
+export const selectSelectedInternalAccountFormattedAddress = createSelector(
   selectSelectedInternalAccount,
   (account) => {
     const selectedAddress = account?.address;
-    return selectedAddress ? toChecksumHexAddress(selectedAddress) : undefined;
+    if (selectedAddress) {
+      // Ethereum accounts should always be checksummed
+      if (isEthAccount(account)) {
+        return toChecksumHexAddress(selectedAddress);
+      }
+      return selectedAddress;
+    }
+    return undefined;
```

Comment:
We could also use the new helper here like:

```suggestion
    // We check for `account` here too, to make sure it will be defined
    // when calling `getFormattedAddressFromAccount`
    return account && account?.address
      ? getFormattedAddressFromAccount(account)
      : undefined;
```